README.md

```diff
@@ -697,7 +752,7 @@ The function accepts several parameters:
 
 ```javascript
 Utils.localize({
-  key: '',
+  key: "",
```

Comment:
Is there a reason why you change all `'` to `"`? Or just a personal preference? Might it be better to use `"` in documentations? I know most languages will enforce usage for `"`, so I'm fine if we chose to go with double quotes instead of single quote in our codebase.

Reply:
AFAIK, [there is no difference](https://medium.com/javascript-in-plain-english/the-real-difference-between-single-quotes-and-double-quotes-in-javascript-3d00bf720bcd).

However, as I usually do the scripting (e.g. in Bash or Perl), I prefer to use single quotes when I don’t need some kind of substitution (like a `$variable` or, in Perl, a `\n`), but this is not a case in JS.

Still, I think we should unite the usually and create a code style guide in the Wiki.

Reply:
We already use the recommended eslint settings, I don't think it's a matter of creating a new style guide. Here it was just a curiosity. I think we could suggest a style (defined on eslint), but not enforce it, because sometime it's better to use one or another.

Reply:
Well, I think we _should_ enforce in the romcal codebase and documentation—thus it would more professional and consistent.

Reply:
I'm not in favor of enforcing it. "it would be more professional and consistent" said by a non JS-developer 😉 
JavaScript is giving us the possibility and the flexibility to use both (unlike other languages), and we even have backtick now! It allows not to escape single quotes in double quotes, or double quotes in single quotes. Also some configuration files written in JavaScript come with double quote, some with single quote. I don't see it as an issue. And like I said in my first comment, I often see documentation using double quote, and codebase using simple quote, so it was mainly a simple curiosity (FYI all romcal's codebase mostly use single quote without enforcing anything, and it's good like that). I think we can use tools with intelligence, so don't try to enforce everything in this project please 😉

Reply:
> "it would be more professional and consistent" said by a non JS-developer wink

Valid argument! :laughing:

Well, you might even use whitespace (of course, not everywhere) as you wish (e.g. use many or use none), but still it is better to write, for example:

```javascript
for (var i = 0; i < max; i++) {
  // Some code
}
```

than

```javascript
for (var i=0;i<max;i++){
  // Some code
}
```

or even remove the newlines and indentation. Both are valid, but it is better to use the former one.

> I don't see it as an issue.

It is not a bug _per se_, but a _style issue_ (it is inconsistent).

> FYI all romcal's codebase **mostly** use single quote without enforcing anything, and it's good like that [emphasis mine]

Exactly: **mostly**. At least for me, _consistent_ means that we use one thing for one purpose only (cf UNIX philosophy). Like:
- typography apostrophes in strings, not typewriter apostrophes;
- one type of quotes throughout the codebase, its comments and the codeblocks in the docs;
- one type of indentation (spaces);
- one width of indentation per file format (two spaces);
- …

> I think we can use tools with intelligence, so don't try to enforce everything in this project please :wink:

I see your point, however, I don’t agree with you.

Anything, as I am not a coder, you coders (@emagnier and @pejulian) should do as you wish. :smiley: I’ve just wanted to express my thoughts how _I_ think it should be. :smiley:

Reply:
I can see you love too much rules and processes, you should work in a bank!
For the spacing and code indentation a tool like Prettier could help.

But please stop complaining and wants to enforce everything if you want to see me continuing on romcal. We are not on a big project with too many contributors. The last 3 months I especially chose to invest a bit of my time on romcal because it's a small project where we can be creative and work on interesting features. However I can see that I'm spending too much time discussing on pointless rules for this sized and the state of this project. As a developer we also need to breathe, and here this is especially tiresome when it comes from a non-developer.

Reply:
> I can see you love too much rules and processes, you should work in a bank!

Well, yeah, I could. :smiley: I love the form (as opposed to content) and accuracy. :laughing:

> But please stop complaining and wants to enforce everything if you want to see me continuing on romcal. […] However I can see that I'm spending too much time discussing on pointless rules for this sized and the state of this project. As a developer we also need to breathe, and here this is especially tiresome when it comes from a non-developer.

I’m sorry. Still: you (the devs/coders) should do as you wish. As you think the discussions like this are pointless, you should just ignore them and stop commenting (or just asnwer those comments that you won’t comment on them).

On my part, I do try more to resist this kind of comments.